py_env_create

```diff
@@ -3,34 +3,48 @@
 # py_env_create -- setup the python environment in order to use CTSM python tools
 #
 # Simple bash script to setup the python environment for the user so they can run the CTSM
-# python tools using "conda".
+# python tools using "conda" or "mamba".
 #
+set -e
+trap 'echo "Exiting on Ctrl+C"; exit 130' INT
+
+SCRIPT_DIR="$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )"
+cd "${SCRIPT_DIR}"
+
 dir=${0%/*}
 if [ "$dir" = "$0" ];then
   dir="."
 fi
 
-# Check if conda is in your path
-conda --help >& condahelp.txt
-error=$?
-if [ $error != 0 ]; then
-   echo "conda is NOT in your path for the bash shell add it with modules or whatever is required on your system to get it in your path"
-   echo "on Derecho/capser/etc use -- module load conda"
-   echo "on izumi/CGD systems use -- module unload lang/python; module load lang/anaconda/23.11.0/base"
-   echo "For notes on installing on a user system see: https://docs.conda.io/projects/conda/en/latest/user-guide/install/index.html"
-   echo "Error code was $error"
-   cat condahelp.txt
-   rm condahelp.txt
-   exit -1
+# Try falling back to mamba if conda doesn't work for some reason
+condamamba="conda"
+tmp_env_name="tmp_env_$(date "+%Y%m%d%H%M%S")"
+set +e
+${condamamba} create --name ${tmp_env_name} --yes --quiet 1>/dev/null 2>&1
```

Comment:
On izumi, I did experience a thing where this was failing, and not because conda wasn't available. So I wonder about allowing more logging around this. But, this is also where the user should be able to go in and fiddle with it.